Clean up regex

50Wliu opened the pull request:
:art: Clean up some really messy regex :art:

Before: `s(h(ift|ow(Mod(elessDialog|alDialog)|Help)))`
After: `shift|showModelessDialog|showModalDialog|showHelp`
:tada:

I probably missed something or messed up a capture group though, so :eyes: would be really appreciated!

kevinsawicki commented:
This looks good at first glance, although it no longer merges cleanly since #154 was merged.

50Wliu commented:
I'm actually going to clean this up a bit more (some things could probably be regrouped like marginTop, marginRight, margin...).

50Wliu commented:
Fixed the merge conflicts.

Also went ahead and re-combined some things (only if there were a bunch like setTime, etc) and fixed a few bugs I found (eg `setChidl`).

mnquintana commented on grammars/javascript.cson:
```diff
@@ -419,19 +419,19 @@
   }
   {
-    'match': '(?<=\\.)(s(hape|ystemId|c(heme|ope|rolling)|ta(ndby|rt)|ize|ummary|pecified|e(ctionRowIndex|lected(Index)?)|rc)|h(space|t(tpEquiv|mlFor)|e(ight|aders)|ref(lang)?)|n(o(Resize|tation(s|Name)|Shade|Href|de(Name|Type|Value)|Wrap)|extSibling|ame)|c(h(ildNodes|Off|ecked|arset)?|ite|o(ntent|o(kie|rds)|de(Base|Type)?|l(s|Span|or)|mpact)|ell(s|Spacing|Padding)|l(ear|assName)|aption)|t(ype|Bo ...
+    'match': '(?<=\\.)(shape|systemId|scheme|scope|scrolling|standby|start|size|summary|specified|sectionRowIndex|selected|selectedIndex|hspace|httpEquiv|htmlFor|height|headers|href|hreflang|noResize|notations|notationName|noShade|noHref|nodeName|nodeType|nodeValue|noWrap|nextSibling|name|ch|childNodes|chOff|checked|charset|cite|content|cookie|cords|code|codeBase|codeType|cols|colSpan|color|compa ...
```
Huh – this is a bit of an aside, but I just noticed that this line isn't syntax highlighted for some reason.

50Wliu replied:
Long lines on Github share the same limitations as on Atom (although I think the char limit is much longer on Github).

kevinsawicki commented:
Thanks so much for cleaning this up, it will be much easier to maintain going forward :blue_heart:
